posthog/models/filters/mixins/property.py

```diff
@@ -101,11 +101,14 @@ def _parse_property_group_list(self, prop_list: Optional[List]) -> Union[List[Pr
             return cast(List[Property], [])
         has_property_groups = False
         has_simple_properties = False
+
         for prop in prop_list:
             if "type" in prop and "values" in prop:
                 has_property_groups = True
-            else:
```

Comment:
@EDsCODE Empty properties shouldn't be considered simple properties? Not sure if this covers the edge cases correctly

Reply:
hmm let's just filter the `prop_list` before this for loop with something like 
`prop_list = filter(lambda prop: prop, prop_list)`

It'll make this logic more clear. Any empty dict shouldn't be a either type of prop

Reply:
If there's an empty dict, this for loop will set both flags to be true and then return the error. So the empty dict needs to be purged

Reply:
(just seeking context here): When do we get empty dicts, that are valid when sent from the frontend, but purged here?

I'd say there shouldn't be any case where this can happen, and if it is happening, should fix it where its being sent, and keeping the API stricter?

Reply:
I thought so too but at the moment if you click new property and don't fill it out, you'll have an empty dict. This is the case with our current property filters too, it's just handled on the backend. I'd say it should be safe on both frontend and backend

Reply:
brrrrrrr. Okay, thanks.

Reply:
@liyiy thoughts on my suggestion?

Reply:
I actually missed the suggestion itself because there were so many comments, thanks for the reminder lol

Reply:
hmm just tried it out and it doesn't work (results aren't changing at all), I'll take a closer look at cleaning this up in the back in a follow up PR